File: server/auth-system/src/main/java/com/example/auth_system/entity/Article.java
package com.example.auth_system.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "Articles")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Article {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int articleID;

    @ManyToOne
    @JoinColumn(name = "createdBy", nullable = false)
    private User createdBy;

    private String articleName;
    private String description;
    private String category;
    private Integer durationMinutes;
    private LocalDateTime creationDate;
    private String imageCover;
    private String content;
}

File: client/src/types/Article.ts
export type Article = {
  articleID: number;
  articleName: string;
  description: string;
  category: string;
  durationMinutes: number;
  imageCover: string;
  content: string;
  creationDate: string;
  authorName?: string; // 
};

export interface ArticleFormProps {
  mode: 'create' | 'edit';
  article?: Article;
  onSuccess: () => void;
  onCancel: () => void;
}
